### ISSUE TITLE
Server crashes after put UAC registrant data

### ISSUE THREAD

**Issue by wiuwiu** [pergolafabio/Hikvision-Addons#13]
Hi.

I have successfully installed your integration in a seperate docker container and successfully linked it to the mariadb container. Opensips works but when I put UAC Registrant data based on your documentation server just crashed. After restart it work for about 1 minute and crashed again.

I installed it on my other machine with HassOS virtual machine and situation is the same.

Here are my logs:
`Apr 22 00:04:06 [51] CRITICAL:core:sig_usr: segfault in process pid: 51, id: 5
Apr 22 00:04:07 [45] INFO:core:handle_sigs: child process 51 exited by a signal 11
Apr 22 00:04:07 [45] INFO:core:handle_sigs: core was generated
Apr 22 00:04:07 [45] INFO:core:handle_sigs: terminating due to SIGCHLD
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 9(55) [TCP receiver] terminated, still waiting for 16 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 10(56) [TCP receiver] terminated, still waiting for 15 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 18(64) [TCP main] terminated, still waiting for 14 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 14(60) [TCP receiver] terminated, still waiting for 13 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 17(63) [Timer handler] terminated, still waiting for 12 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 15(61) [TCP receiver] terminated, still waiting for 11 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 12(58) [TCP receiver] terminated, still waiting for 10 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 13(59) [TCP receiver] terminated, still waiting for 9 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 16(62) [TCP receiver] terminated, still waiting for 8 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 1(47) [HTTPD INADDR_ANY:8888] terminated, still waiting for 7 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 11(57) [TCP receiver] terminated, still waiting for 6 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 2(48) [MI FIFO] terminated, still waiting for 5 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 6(52) [SIP receiver udp:192.168.30.50:5060] terminated, still waiting for 4 more
Apr 22 00:04:07 [50] INFO:core:sig_usr: signal 15 received
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 4(50) [timer] terminated, still waiting for 3 more
Apr 22 00:04:07 [49] INFO:core:sig_usr: signal 15 received
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 3(49) [time_keeper] terminated, still waiting for 2 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 8(54) [SIP receiver udp:192.168.30.50:5060] terminated, still waiting for 1 more
Apr 22 00:04:07 [45] INFO:core:shutdown_opensips: process 7(53) [SIP receiver udp:192.168.30.50:5060] terminated, still waiting for 0 more
Apr 22 00:04:07 [45] INFO:core:cleanup: cleanup
Apr 22 00:04:07 [45] NOTICE:presence:destroy: destroy module ...
Thank you for running opensips`

**Comment by pergolafabio** [comment]
Strange, the log is not indicating any usefull error... So this only happens after inserting uac? If you insert other stuff like the domain or user , that works?

Can you send a screenshot of the uac you enter?

**Comment by wiuwiu** [comment]
Domain and user stuff works very well and using softphone works perfect. 
The only problem is UAC Registrant - server crashes when I chance UAC Registrant status to "Enable".

**Comment by pergolafabio** [comment]
Strange.... Can you make screenshot? Try inputting the row maybe manually in Mariadb? Or enable it in SQL?

**Comment by wiuwiu** [comment]
I have DS-KH8350-WTE1/EU indoor station with these settings:
```
Device Type: Indoor Extension
Serial No.: Q12345678
IP Address: 192.168.30.20 - this is opensips server
Gateway: 192.168.30.254
Subnet Mask: 255.255.255.0
Password: XXXXX
SIP No. 10000000005
```

Here are my opensips server settings:
```
Registrar Server: sip:192.168.150.12:5065 - this is indoor station
Address-of-Record: sip:10000000005@192.168.150.12:5065
Auth Username: 10000000005
Auth Password: XXXXX
Contact URI: sip:10000000005@192.168.30.20
```

Row in registrant table is created automatically.

**Comment by pergolafabio** [comment]
That looks fine indeed... Try inserting that row manually?
No idea what could be wrong :-(

**Comment by wiuwiu** [comment]
It doesn't matter how I put UAC Registrant data. When UAC Registrant is enabled after some time (about 1 minute) server crashes.

**Comment by wiuwiu** [comment]
One more thing, calling from indoor panels to softphone works very well.

**Comment by pergolafabio** [comment]
Can you launch sngrep ? After adding UAC, I think you will see a flooding of messages... That will kill the server, I had it too...

Can you check the type of sip message there?

**Comment by wiuwiu** [comment]
There is no data in sngrep.
But I have a question, what setting do you have in Network -> SIP Server Configuration and Network -> Group Network Settings.

**Comment by pergolafabio** [comment]
Ok, will have a look next week when I'm home

**Comment by pergolafabio** [comment]
Hey, where do you see "network"? i dont have any network settings in opensips?


![image](https://user-images.githubusercontent.com/43251136/165067386-0dd6ccfc-fc18-47e2-baff-ad6902a60c8b.png)

**Comment by wiuwiu** [comment]
Not in opensips but in your indoor video intercom - but I think these settings are not responsible for opensips crashing. This must be something in opensips. 

Please have a look at my logs. These are my server starting logs:
```
Apr 25 13:15:47 [44] CRITICAL:core:set_core_dump: core limits increased only to 0
Apr 25 13:15:47 [44] NOTICE:core:main: version: opensips 3.2.6 (x86_64/linux)
Apr 25 13:15:47 [44] NOTICE:core:main: using 32 MB of shared memory, allocator: F_MALLOC
Apr 25 13:15:47 [44] NOTICE:core:main: using 16 MB of private process memory, allocator: F_MALLOC
Apr 25 13:15:47 [44] INFO:core:init_reactor_size: reactor size 40960 (using up to 1.56Mb of memory per process)
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_CORE_THRESHOLD(0)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_CORE_SHM_THRESHOLD(1)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_CORE_PKG_THRESHOLD(2)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_CORE_PROC_AUTO_SCALE(3)>
Apr 25 13:15:47 [44] INFO:tm:mod_init: TM - initializing...
Apr 25 13:15:47 [44] INFO:sl:mod_init: Initializing StateLess engine
Apr 25 13:15:47 [44] NOTICE:signaling:mod_init: initializing module ...
Apr 25 13:15:47 [44] INFO:uac_registrant:load_reg_info_from_db: loading [1] records from db
Apr 25 13:15:47 [44] INFO:rr:mod_init: rr - initializing
Apr 25 13:15:47 [44] INFO:maxfwd:mod_init: initializing...
Apr 25 13:15:47 [44] INFO:sipmsgops:mod_init: initializing...
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_MYSQL_CONNECTION(4)>
Apr 25 13:15:47 [44] INFO:usrloc:ul_init_locks: locks array size 512
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_UL_AOR_INSERT(5)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_UL_AOR_DELETE(6)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_UL_CONTACT_INSERT(7)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_UL_CONTACT_UPDATE(8)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_UL_CONTACT_DELETE(9)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_UL_CONTACT_REFRESH(10)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_UL_LATENCY_UPDATE(11)>
Apr 25 13:15:47 [44] INFO:registrar:mod_init: initializing...
Apr 25 13:15:47 [44] INFO:acc:mod_init: initializing...
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_ACC_EVENT(12)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_ACC_CDR(13)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_ACC_MISSED_EVENT(14)>
Apr 25 13:15:47 [44] INFO:auth:mod_init: initializing...
Apr 25 13:15:47 [44] INFO:auth_db:mod_init: initializing...
Apr 25 13:15:47 [44] INFO:alias_db:mod_init: initializing...
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_PRESENCE_PUBLISH(15)>
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_PRESENCE_EXPOSED(16)>
Apr 25 13:15:47 [44] INFO:dialplan:mod_init: initializing module...
Apr 25 13:15:47 [44] INFO:drouting:dr_init: dynamic routing - initializing
Apr 25 13:15:47 [44] INFO:drouting:init_prefix_tree: counted 10 possible chars under a node
Apr 25 13:15:47 [44] INFO:core:evi_publish_event: Registered event <E_DROUTING_STATUS(17)>
Apr 25 13:15:47 [44] INFO:core:mod_init: initializing UDP-plain protocol
Apr 25 13:15:47 [44] INFO:core:mod_init: initializing TCP-plain protocol
Apr 25 13:15:47 [49] WARNING:dialplan:dp_load_db: no data in the db
Apr 25 13:15:47 [49] INFO:drouting:dr_reload_data_head: loading drouting data!
Apr 25 13:15:47 [49] NOTICE:drouting:dr_load_routing_info: loaded 0 gateways in partition 'Default'
Apr 25 13:15:47 [49] NOTICE:drouting:dr_load_routing_info: loaded 0 carriers in partition 'Default'
Apr 25 13:15:47 [49] NOTICE:drouting:dr_load_routing_info: loaded 0 rules in partition 'Default'
```
And these are my logs after I enable UAC Registar and hit 'Reload on server':
```
Apr 25 13:17:22 [45] INFO:uac_registrant:load_reg_info_from_db: loading [1] records from db
Apr 25 13:17:27 [49] CRITICAL:core:sig_usr: segfault in process pid: 49, id: 5
Apr 25 13:17:27 [44] INFO:core:handle_sigs: child process 49 exited by a signal 11
Apr 25 13:17:27 [44] INFO:core:handle_sigs: core was not generated
Apr 25 13:17:27 [44] INFO:core:handle_sigs: terminating due to SIGCHLD
Apr 25 13:17:27 [47] INFO:core:sig_usr: signal 15 received
Apr 25 13:17:27 [48] INFO:core:sig_usr: signal 15 received
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 1(45) [HTTPD INADDR_ANY:8888] terminated, still waiting for 16 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 2(46) [MI FIFO] terminated, still waiting for 15 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 3(47) [time_keeper] terminated, still waiting for 14 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 4(48) [timer] terminated, still waiting for 13 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 6(50) [SIP receiver udp:192.168.30.20:5060] terminated, still waiting for 12 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 9(53) [TCP receiver] terminated, still waiting for 11 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 10(54) [TCP receiver] terminated, still waiting for 10 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 11(55) [TCP receiver] terminated, still waiting for 9 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 14(58) [TCP receiver] terminated, still waiting for 8 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 13(57) [TCP receiver] terminated, still waiting for 7 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 8(52) [SIP receiver udp:192.168.30.20:5060] terminated, still waiting for 6 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 18(62) [TCP main] terminated, still waiting for 5 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 16(60) [TCP receiver] terminated, still waiting for 4 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 7(51) [SIP receiver udp:192.168.30.20:5060] terminated, still waiting for 3 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 12(56) [TCP receiver] terminated, still waiting for 2 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 15(59) [TCP receiver] terminated, still waiting for 1 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 17(61) [Timer handler] terminated, still waiting for 0 more
Apr 25 13:17:27 [44] INFO:core:cleanup: cleanup
Apr 25 13:17:27 [44] NOTICE:presence:destroy: destroy module ...
Thank you for running opensips
Apr 25 13:17:27 [49] CRITICAL:core:sig_usr: segfault in process pid: 49, id: 5
Apr 25 13:17:27 [44] INFO:core:handle_sigs: child process 49 exited by a signal 11
Apr 25 13:17:27 [44] INFO:core:handle_sigs: core was not generated
Apr 25 13:17:27 [44] INFO:core:handle_sigs: terminating due to SIGCHLD
Apr 25 13:17:27 [47] INFO:core:sig_usr: signal 15 received
Apr 25 13:17:27 [48] INFO:core:sig_usr: signal 15 received
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 1(45) [HTTPD INADDR_ANY:8888] terminated, still waiting for 16 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 2(46) [MI FIFO] terminated, still waiting for 15 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 3(47) [time_keeper] terminated, still waiting for 14 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 4(48) [timer] terminated, still waiting for 13 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 6(50) [SIP receiver udp:192.168.30.20:5060] terminated, still waiting for 12 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 9(53) [TCP receiver] terminated, still waiting for 11 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 10(54) [TCP receiver] terminated, still waiting for 10 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 11(55) [TCP receiver] terminated, still waiting for 9 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 14(58) [TCP receiver] terminated, still waiting for 8 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 13(57) [TCP receiver] terminated, still waiting for 7 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 8(52) [SIP receiver udp:192.168.30.20:5060] terminated, still waiting for 6 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 18(62) [TCP main] terminated, still waiting for 5 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 16(60) [TCP receiver] terminated, still waiting for 4 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 7(51) [SIP receiver udp:192.168.30.20:5060] terminated, still waiting for 3 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 12(56) [TCP receiver] terminated, still waiting for 2 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 15(59) [TCP receiver] terminated, still waiting for 1 more
Apr 25 13:17:27 [44] INFO:core:shutdown_opensips: process 17(61) [Timer handler] terminated, still waiting for 0 more
Apr 25 13:17:27 [44] INFO:core:cleanup: cleanup
Apr 25 13:17:27 [44] NOTICE:presence:destroy: destroy module ...
Thank you for running opensips
```

**Comment by pergolafabio** [comment]
ah ok, yeah i dont have any SIP settings configred on my hikvision hardware, thats indeed the purpose of this addon, so thats not needed

looks that error is quite common on opensips, google this one:

opensips uac :core:sig_usr: segfault in process pid: 49, id: 5

i dont have control over this error, i just use the opensips default install guide

**Comment by wiuwiu** [comment]
I tested it on two seperate machines - one with Home Assistant Container as a seperate docker container and second with Home Assistant OS as a integration.

**Comment by pergolafabio** [comment]
strange, i dont have any clue, as you can see in dockerfile, i just follow the manual to install opensips, alle latest modules
it works for other people running HassOS , also for me its stable since the first day i released it

**Comment by pergolafabio** [comment]
can you run snggrep inside the opensips docker container? ot within the SSH addon , by installing "apk add sngrep"

i think the opensips crashes by a message flood, returned by your indoor, i had it too in beginning, with my 9310 as indoor station...

so i remove that message from opensips.cfg to stop flooding

**Comment by wiuwiu** [comment]
I don't have any data in snggrep when I set UAC Registar to enabled. In other cases, like using softphone ,there are some data but in this case there is no data.

**Comment by wiuwiu** [comment]
I think I will try run opensips with mariadb in one container.

**Comment by pergolafabio** [comment]
hmm, ok, then the flood is not related, no idea :-(

you can use any sql you want, just change it also in the cfg file that i include, its on several locations

**Comment by pergolafabio** [comment]
huh, for some reason, i know have the exact same problem
i did an update on my addon, nothing really changed actually, now it crashes too when i enable the UAC

```
Apr 26 22:36:29 [47] INFO:uac_registrant:load_reg_info_from_db: loading [1] records from db
Apr 26 22:38:39 [47] INFO:uac_registrant:load_reg_info_from_db: loading [1] records from db
Apr 26 22:38:48 [61] CRITICAL:core:sig_usr: segfault in process pid: 61, id: 15
Apr 26 22:38:49 [45] INFO:core:handle_sigs: child process 61 exited by a signal 11
Apr 26 22:38:49 [45] INFO:core:handle_sigs: core was generated
Apr 26 22:38:49 [45] INFO:core:handle_sigs: terminating due to SIGCHLD
```

have you found out?

**Comment by pergolafabio** [comment]
i also see this on my mariadb addon, not sure if its related:

```
2022-04-26 22:38:49 343 [Warning] Aborted connection 343 to db: 'opensips' user: 'opensips' host: '172.30.32.1' (Got an error reading communication packets)
2022-04-26 22:38:49 346 [Warning] Aborted connection 346 to db: 'opensips' user: 'opensips' host: '172.30.32.1' (Got an error reading communication packets)
2022-04-26 22:38:49 344 [Warning] Aborted connection 344 to db: 'opensips' user: 'opensips' host: '172.30.32.1' (Got an error reading communication packets)
2022-04-26 22:38:49 342 [Warning] Aborted connection 342 to db: 'opensips' user: 'opensips' host: '172.30.32.1' (Got an error reading communication packets)
2022-04-26 22:38:49 355 [Warning] Aborted connection 355 to db: 'opensips' user: 'opensips' host: '172.30.32.1' (Got an error reading communication packets)
```

**Comment by pergolafabio** [comment]
wait, seems there was a new release of 3.2.6 , released 22/04 
i think when i updated my addon, opensips also upgraded from 3.2.5 to 3.2.6

and you started installing just a few days ago, so your docker already downloaded 3.2.6 ....
that must be it ....

https://www.opensips.org/About/Version-3-2-x

**Comment by pergolafabio** [comment]
BACK up and running, i reverted back to 3.2.5 , now its working again!! omg , what a relief :-)

**Comment by pergolafabio** [comment]
created bug rebport here : 

https://github.com/OpenSIPS/opensips/issues/2823

**Comment by wiuwiu** [comment]
Hi. I revert back to 3.2.5 version as you mentioned. There is no opensips crash :) and I have a "Registered" extension status in my indoor station. But when I try to make a call from indoor station to "opensips" server I get this in my logs:
`ERROR:uac_registrant:run_reg_tm_cback: No contact header in received 200ok in state [2]` and after about 5 s call is terminated.

**Comment by pergolafabio** [comment]
Good to hear!! ignore those messages, i have them too:

```
Apr 28 16:50:26 [52] ERROR:uac_registrant:run_reg_tm_cback: No contact header in received 200ok in state [2]
Apr 28 16:52:07 [53] ERROR:uac_registrant:run_reg_tm_cback: No contact header in received 200ok in state [2]
Apr 28 16:53:46 [51] ERROR:uac_registrant:run_reg_tm_cback: No contact header in received 200ok in state [2]
Apr 28 16:55:27 [50] ERROR:uac_registrant:run_reg_tm_cback: No contact header in received 200ok in state [2]
Apr 28 16:57:06 [52] ERROR:uac_registrant:run_reg_tm_cback: No contact header in received 200ok in state [2]
```

But you cant call from indoorstation to opensips / asteriks trunk, that doesnt work some for some reason
Try calling from outdoor, then it works, when you registered a softphone to opensips, or if you are using asterisk...

keep me posted

**Comment by wiuwiu** [comment]
What do you mean "Try calling from outdoor"? You mean from main outdoor station?
As for softphone... I don't need to setup UAC registrant to call from softphone. It's enough to setup SIP server on indoor station and point to opensips to have a possibility calling from softphone to indoor station and from indoor station to softphone. When I calling from softphone (linphone on android) I have a video from my mobile on indoor station.

**Comment by pergolafabio** [comment]
I dont understand your setup, you already have your indoor/outdoor devices setup to a PBX ?
Why are you using this addon then? The reason for this addon, is that you dont need a PBX in the foreground, so no need for your outdoor/indoor register to a SIP server...
Because if you setup outdoor/indoor to SIP, you loose early video on your indoor stations

This addon, gives the possibility to workaround, so you can register softphones to opensips, so when you call from outdoor to indoor, you still see early video on indoor stations, also hikconnect cloud still works, and you can register a softphone you like, no need for hikconnect app anymore...

you are mixing both stuff?

**Comment by pergolafabio** [comment]
because if you just point your indoor to a opensips or another PBX, when you then have an incoming call, the softphone will not ring... because its not an extension...

The only advantage is when you register your indoor to a PBX, is that you can call the indoor station from any client, but i dont see any advantage in that ....

i want a good intercom solututon , with early media on my hikvision indoor screens, and have the flexibility to have softphones of any choice, like linphone or HA lovelace card, thats the usecase of this addon :-)

**Comment by wiuwiu** [comment]
@pergolafabio what do you mean "early video"? I point my indoor station to opensips server and I have a video when I call from softphone to indoor station and from outdoor to indoor station.

**Comment by pergolafabio** [comment]
How do you point your indoor station to opensips? you created a user on Opensips server? but then what? whats the purpose? you can only make calls between indoor and softphones? thats all, right?

**Comment by pergolafabio** [comment]
if you just want to call from indoor stations, to softphones and visa versa, then you dont need this addon, you can just use asterisk... much easier then opensips

I just created opensips because of the UAC registant, i send a custom XML header... thats a feature not present in Asterisk (trunk) ... so i just fake a new indoor station, registered as extension in primary panel....  :-)

**Comment by wiuwiu** [comment]
> yes, because if you call from outdoor to indoor, you use indeed hikvision SIP protocol But if you point your indoor station to a PBX, like in this case Opensips, if you call from outdoor station, your indoor station will indeed see early video (video before pickup) , but your softphone will not ring.... your indoor station doesnt forward the call to a softphone ... or is it?
> 
> Normally only extentensions registered to an indoor station will only ring when you call from the outdoor station

Yes and no :) to have a call forwarding from my indoor to softphone I put my softphone sip number into outdoor station into Intercom -> Number settings. So when i call from outdoor first indoor station is calling, then I have a record message prompt and after that my softphone is calling.



> How do you point your indoor station to opensips? you created a user on Opensips server? but then what? whats the purpose? you can only make calls between indoor and softphones? thats all, right?

Yes.

**Comment by pergolafabio** [comment]
yes;, then you have indeed early media ... but  i dont want call forwarding

i want my indoor stations ring at same time with early media
my softphones should ring too
and i dont want to use the SIP protocol on my indoor/outdoor station, too buggy

**Comment by pergolafabio** [comment]
Do you also have regusteried your outdoor server to a PBX? because in number settins, what have you put there ? a SIP nr right? so you outdoor station is also using a PBX?

The best soluton , was to have a setting that it could call indoor station AND at the same time also SIP numbers, that would be great!! like a multicall, then all this stuff was not needed

**Comment by wiuwiu** [comment]
> yes;, then you have indeed early media ... but i dont want call forwarding
> 
> i want my indoor stations ring at same time with early media my softphones should ring too and i dont want to use the SIP protocol on my indoor/outdoor station, too buggy

OK, but how to do this?



> Do you also have regusteried your outdoor server to a PBX? because in number settins, what have you put there ? a SIP nr right? so you outdoor station is also using a PBX?

No, I don't point my outdoor station to opensips but only indoor station is connected to opensips. I think outdoor station can inform indoor station about sip number to call.

**Comment by pergolafabio** [comment]
Ah, so that means your indoor station is doing the forward I think....

Well, if you want the multi call, so your indoor rings AND same time your sofphone ... Then you don't need to setup your indoor to opensips, remove those sip settings... You need to setup the UAC, that UAC forwards the call to the user in opensips...

So on doorpress event, your indoor rings and the UAC receives call to, that forwards to your sofphone.....

**Comment by pergolafabio** [comment]
Yes, indeed, no video here either, I have that issue too, but I'm working on an RTSP inject tool :-)

**Comment by pergolafabio** [comment]
I don't understand either why I don't have video, I did a trace and everything looks good... I think the indoor I'd publishing the video with rtsp stream to the secundaire extensions... Not sure , I need to trace that first with a real indoor monitor

**Comment by wiuwiu** [comment]
So you don't have a video on softphone in multicall sesion too?

**Comment by pergolafabio** [comment]
Indeed...

But I'm using HA companion app to receive calls... So I can use the camera card :-)

**Comment by pergolafabio** [comment]
But I'm working on it, I maybe have found an RTSP inject tool

**Comment by wiuwiu** [comment]
OK, I will try your solution :)

**Comment by pergolafabio** [comment]
Gonna leave this open, untill get feedback from opensips about this issue in 3.2.6

**Comment by wiuwiu** [comment]
Sure.

**Comment by pergolafabio** [comment]
Hey @wiuwiu , it's fixed in the opensips nightly build...
Fix will be present also in 3.2.7
